scilpy/tractanalysis/reproducibility_measures.py

```diff
@@ -395,3 +410,248 @@ def compute_dice_streamlines(bundle_1, bundle_2):
+
+        global B
+        has_data = sh_data_1[vox_ind].any() and sh_data_2[vox_ind].any()
+        if has_data:
+            sf_1 = np.dot(sh_data_1[vox_ind], B)
+            sf_2 = np.dot(sh_data_2[vox_ind], B)
+            acc = np.corrcoef(sf_1, sf_2)[0, 1]
+        else:
+            acc = np.nan
+
+        if skip_streamlines_distance:
+            results.append([np.nan, acc])
+            continue
+
+        # Get the streamlines in the neighborhood (i.e., 1.5mm away)
+        pts_ind_1 = tree_1.query_ball_point(vox_ind, 1.5)
+        if not pts_ind_1:
+            results.append([np.nan, acc])
+            continue
+        strs_ind_1 = np.unique(matched_points_1[pts_ind_1])
+        neighb_streamlines_1 = sft_1.streamlines[strs_ind_1]
+
+        # Get the streamlines in the neighborhood (i.e., 1.5mm away)
+        pts_ind_2 = tree_2.query_ball_point(vox_ind, 1.5)
+        if not pts_ind_2:
+            results.append([np.nan, acc])
+            continue
+        strs_ind_2 = np.unique(matched_points_2[pts_ind_2])
+        neighb_streamlines_2 = sft_2.streamlines[strs_ind_2]
+
+        # Using neighb_streamlines (all streamlines in the neighborhood of our
+        # voxel), we can compute the distance between the two sets of
+        # streamlines using FSS (FastStreamlineSearch).
+        with warnings.catch_warnings(record=True) as _:
+            fss = FastStreamlineSearch(neighb_streamlines_1, 10, resampling=12)
+            dist_mat = fss.radius_search(neighb_streamlines_2, 10)
+            sparse_dist_mat = np.abs(dist_mat.tocsr()).toarray()
+            sparse_ma_dist_mat = np.ma.masked_where(sparse_dist_mat < 1e-3,
+                                                    sparse_dist_mat)
+            sparse_ma_dist_vec = np.squeeze(np.min(sparse_ma_dist_mat,
+                                                   axis=0))
+
+            # dists will represent the average distance between the two sets of
+            # streamlines in the neighborhood of the voxel.
+            dist = np.average(sparse_ma_dist_vec)
+            results.append([dist, acc])
+
+    return results
+
+
+def _compare_tractogram_wrapper(mask, nbr_cpu, skip_streamlines_distance):
+    """
+    Wrapper for the comparison of two tractograms. This function uses
+    multiprocessing to compute the difference between two sets of streamlines
+    for each voxel.
+
+    This function simply calls the function _compute_difference_for_voxel(),
+    which expect chunks of indices to process and use global variables to avoid
+    duplicating the data for each chunk of voxels.
+
+    Use the function tractogram_pairwise_comparison() as an entry point.
+
+    Parameters
+    ----------
+    mask: np.ndarray
+        Mask of the data to compare.
+    nbr_cpu: int
+        Number of CPU to use.
+    skip_streamlines_distance: bool
+        If true, skip the computation of the distance between streamlines.
+
+    Returns
+    -------
+    Tuple of np.ndarray
```

Comment:
We don't usually put it like this. By default, python always returns tuples, no need to say it. You can remove this line and unindent the two below.